examples/blaze/src/main/scala/com/example/http4s/blaze/demo/server/Module.scala

```diff
@@ -17,7 +17,10 @@ import org.http4s.server.middleware.{AutoSlash, ChunkAggregator, GZip, Timeout}
 
 import scala.concurrent.duration._
 
-class Module[F[_]](client: Client[F])(implicit F: ConcurrentEffect[F], S: Scheduler, T: Timer[F]) {
+class Module[F[_]](client: Client[F])(
+    implicit F: ConcurrentEffect[F],
+    S: Scheduler,
+    T: Timer[OptionT[F, ?]]) {
```

Comment:
Sorry, is there a reason this is necessary in particular to be `OptionT`? I'm missing something not immediately clear upon inspection.

Also, `Timer` is there to replace fs2 scheduler eventually. Maybe in another PR we can proceed to clean up examples, but when you have one, the other is no longer needed

Reply:
This is `Timeout` pre-generalisation:

```scala
  def apply[F[_]](timeout: FiniteDuration, timeoutResponse: F[Response[F]])(
      @deprecatedName('service) routes: HttpRoutes[F])(
      implicit F: Concurrent[F],
      T: Timer[F]): HttpRoutes[F] =???
```

`F` is the non-`OptionT`-adorned `F` that parameterises `HttpRoutes[F]` (which expands to `Http[OptionT[F, ?], F]`), so it doesn't matter that there's no `Timer[OptionT[F, ?]]` because we don't require evidence for one.

But by having now generalised it like so:

```scala
  def apply[F[_], G[_], A](timeout: FiniteDuration, timeoutResponse: F[Response[G]])(
      @deprecatedName('service) http: Kleisli[F, A, Response[G]])(
      implicit F: Concurrent[F],
      T: Timer[F]): Kleisli[F, A, Response[G]] = ???
```

... then when `F` is instantiated within `Module` (and I guess most places `Timeout` is used in userland too), it becomes `OptionT[F, ?]`, and we now require evidence of `Timer[OptionT[F, ?]]`:

```scala
  private val timeoutHttpEndpoint: HttpRoutes[F] =
    new TimeoutHttpEndpoint[F].service

  private val timeoutEndpoints: HttpRoutes[F] =
    Timeout(1.second)(timeoutHttpEndpoint)
```

This is quite gnarly though (and what prompted me to `Timer.derive[F]` at the use-site previously. ) Is there a simpler solution that I'm not seeing? I guess the real fix would be to have transformer instances for `Timer` defined upstream in cats.

Reply:
Nope, 100% my bad I misread this. I just thought `Timeout` wasn't being used at all and/or it was being passed a `Http[F, F]`. You're right :+1:

Reply:
Cool. Happy to tackle the clean up of examples in a separate PR, as you suggest 😄